src/main/java/ttt/gui/GuiGameController.java

```diff
@@ -5,53 +5,85 @@
 
 import java.util.List;
 
+import static ttt.player.GuiHumanPlayer.IGNORE_AS_MOVE_WILL_COME_FROM_DISPLAY;
+
 public class GuiGameController implements GameController {
 
+    private final GameConfiguration gameConfiguration;
     private GameRules ticTacToeRules;
     private DisplayPresenter boardView;
+    private GameType gameType;
 
-    public GuiGameController(GameRules ticTacToeRules, ViewFactory viewFactory) {
+    public GuiGameController(GameConfiguration gameConfiguration, GameRules ticTacToeRules, ViewFactory viewFactory) {
+        this.gameConfiguration = gameConfiguration;
         this.ticTacToeRules = ticTacToeRules;
         this.boardView = viewFactory.createView(this, ticTacToeRules);
     }
 
     @Override
     public void presentGameTypes() {
-        List<GameType> allGameTypes = ticTacToeRules.getGameTypes();
+        List<GameType> allGameTypes = gameConfiguration.getGameTypes();
         boardView.presentGameTypes(allGameTypes);
     }
 
     @Override
     public void presentBoardDimensionsFor(GameType gameType) {
-        ticTacToeRules.storeGameType(gameType);
-        String dimension = ticTacToeRules.getDimension(gameType);
+        setGameType(gameType);
+        String dimension = gameConfiguration.getDimension(gameType);
         boardView.presentGridDimensionsUpTo(dimension);
     }
 
     @Override
     public void presentBoard(String dimension) {
-        ticTacToeRules.initialiseGame(dimension);
+        ticTacToeRules.initialiseGame(gameType, dimension);
         Board board = ticTacToeRules.getBoard();
+
+        playAutomatedMoveIfAppropriate();
         boardView.presentsBoard(board);
     }
 
     @Override
     public void playMove(String position) {
-        ticTacToeRules.playMoveAt(position);
-        Board board = ticTacToeRules.getBoard();
-        boardView.presentsBoard(board);
+        playMoveIfSpaceOnBoard(Integer.valueOf(position));
+        playAutomatedMoveIfAppropriate();
```

Comment:
Previously I had the below, which also felt like a OCP violation. I've tried to address that here, again with the newly introduced GuiHumanPlayer. This way I don't have to query the 
GameType to make a decision on whether an automated move is required or not. 

' if ((!gameType.equals(HUMAN_VS_HUMAN)) && gameInProgress()) {
        ..play automated move
 }'